docs/CMakeLists.txt

```diff
@@ -22,3 +22,10 @@ add_custom_target(docs
   COMMENT "Generating documentation"
   VERBATIM
 )
+
+foreach(module IN LISTS simple_modules)
+  file(WRITE "module/${module}.rst" "\
+.. default-domain:: cmake
+.. cmake-module:: ../../src/modules/${module}.cmake
+")
+endforeach()
```

Comment:
This I do not like. It writes to the source tree unconditionally on every configure. Modifying the source tree should be manually triggered. Either make it a manual button, or possibly even better, generate those files in the build tree during building of the `docs` target.

Reply:
Rewritten to perform a check and throw errors. As discussed.